### PR TITLE
Validating informational fields against effective schema

### PR DESCRIPTION
**Pull request by thomasabraham**
# Validating informational fields (ordering_key_fields and ordering_instance_ids) against effective schema.

> Zalando ticket : [ARUHA-2048](https://jira.zalando.net/browse/ARUHA-2048)

## Description
Both ordering_key_fields and ordering_instance_ids are validated against the user-provided schema during event type creation, but the documentation (and the API guild) say that it should be validated against the effective schema, such that people could reference the metadata object, for example.

AC:
Both fields validated against the effective schema during event type creation.

## Review
- [ ] Tests
- [ ] Documentation

## Deployment Notes
These should highlight any db migrations, feature toggles, etc.

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/EventTypeService.java [hunk 1]
```diff
@@ -101,6 +102,7 @@
     private final String etLogEventType;
     private final NakadiAuditLogPublisher nakadiAuditLogPublisher;
     private final EventTypeOptionsValidator eventTypeOptionsValidator;
+    private final JsonSchemaEnrichment jsonSchemaEnrichment = new JsonSchemaEnrichment();
```
Comment (Kunal-Jha) [inline src/main/java/org/zalando/nakadi/service/EventTypeService.java]: Isn't it better to add the initialization to the constructor?

File: src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java [hunk 2]
```diff
@@ -215,6 +218,16 @@ public void testFeatureToggleForbidsDeleteEventTypeWithNonDeletableSubscription(
...
     }
 
+    @Test
+    public void testAllowCreatingEventTypeWithInformationalFieldsFromEffectiveSchema(){
+        final EventType et = EventTypeTestBuilder.builder()
+                .category(EventCategory.DATA)
+                .build();
+        et.setOrderingKeyFields(Collections.singletonList("metadata.occurred_at"));
+        et.setOrderingInstanceIds(Collections.singletonList("metadata.partition"));
+        eventTypeService.create(et, true);
```
Comment (Kunal-Jha) [inline src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java]: Shouldn't the test have some validation step?

**Comment by Kunal-Jha** [comment]
@thomasabraham looks good. 👍  some minor comments. Also please change the description to reflect the things changed in the PR. Alos adding a link to the ticket is usually helpful

File: src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java (hunk 2 above)
Comment (thomasabraham) [reply src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java]: Before the fix, the test will break with exception.

File: src/main/java/org/zalando/nakadi/service/EventTypeService.java (hunk 1 above)
Comment (thomasabraham) [reply src/main/java/org/zalando/nakadi/service/EventTypeService.java]: Done

File: src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java (hunk 2 above)
Comment (thomasabraham) [reply src/test/java/org/zalando/nakadi/service/EventTypeServiceTest.java]: Okay I got the point; fixed.

**Comment by thomasabraham** [comment]
Screwed up the branch; will create a new branch.

**Pull request closed without merging**